crates/autopilot/src/periodic_db_cleanup.rs

```diff
@@ -0,0 +1,134 @@
+use {
+    crate::database::Postgres,
+    chrono::{DateTime, Utc},
+    std::time::Duration,
+    tokio::time,
+};
+
+pub struct OrderEventsCleanerConfig {
+    cleanup_interval: Duration,
+    event_age_threshold: chrono::Duration,
+}
+
+impl OrderEventsCleanerConfig {
+    pub fn new(cleanup_interval: Duration, event_age_threshold: Duration) -> Self {
+        OrderEventsCleanerConfig {
+            cleanup_interval,
+            event_age_threshold: chrono::Duration::from_std(event_age_threshold).unwrap(),
+        }
+    }
+}
+
+pub struct OrderEventsCleaner {
+    config: OrderEventsCleanerConfig,
+    db: Postgres,
+}
+
+impl OrderEventsCleaner {
+    pub fn new(config: OrderEventsCleanerConfig, db: Postgres) -> Self {
+        OrderEventsCleaner { config, db }
+    }
+
+    pub async fn run_forever(self) -> ! {
+        let mut interval = time::interval(self.config.cleanup_interval);
+        loop {
+            let timestamp: DateTime<Utc> = Utc::now() - self.config.event_age_threshold;
+            match self.db.delete_events_before(timestamp).await {
+                Ok(affected_rows_count) => {
+                    tracing::debug!(
+                        "deleted {:?} order events before {}",
+                        affected_rows_count,
+                        timestamp
+                    );
+                    Metrics::get()
+                        .last_order_events_cleanup_run
+                        .set(Utc::now().timestamp())
+                }
+                Err(err) => {
+                    tracing::warn!(?err, "failed to delete order events before {}", timestamp)
+                }
+            }
+            interval.tick().await;
```

Comment:
til about `tokio::time::interval`. Are you sure though that the first tick doesn't return immediately (at least that's what I read from their [example](https://docs.rs/tokio/latest/tokio/time/struct.Interval.html#examples). Maybe a simple sleep would also work?

Reply:
`sleep` leads to time drifts, which is not expected behavior for the current logic. Added initial `tick` call instead.

Reply:
nit: why don't we tick().await inside the loop in first place instead of calling it twice?